### PR TITLE
[SPARK-27460][FOLLOW-UP][TESTS] Fix flaky tests

### PR DESCRIPTION
**Pull request by gatorsmile**
## What changes were proposed in this pull request?

This patch makes several test flakiness fixes.

## How was this patch tested?
N/A

### CODE REVIEW

**Comment by gatorsmile** [comment]
cc @gengliangwang

**Comment by gengliangwang** [comment]
LGTM, I did see some of these flaky tests in https://github.com/apache/spark/pull/24373 . Increasing the timeout makes the parallel tests more robust.
I will trigger a round of concurrent Jenkins jobs to see if any flaky tests still existed.

**Comment by HyukjinKwon** [comment]
retest this please

File: resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala [hunk 1]
```diff
@@ -205,15 +205,15 @@ class YarnClusterSuite extends BaseYarnClusterSuite {
       .startApplication()
 
     try {
-      eventually(timeout(30.seconds), interval(100.milliseconds)) {
+      eventually(timeout(180.seconds), interval(100.milliseconds)) {
```
Comment (srowen) [inline resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala]: You can write "3.minutes" in cases like this and others below, but it doesn't matter.

**Review by srowen — APPROVED** [review]

**Review by felixcheung — APPROVED** [review]

**Comment by srowen** [comment]
That's a weird error... it's from javadoc:
```
[error] /home/jenkins/workspace/NewSparkPullRequestBuilder/core/target/java/org/apache/spark/serializer/SerializationDebugger.java:159: error: cannot find symbol
[error]   static private  org.apache.spark.serializer.SerializationDebugger.ObjectStreamClassReflection reflect ()  { throw new RuntimeException(); }
[error]                                                                    ^
[error]   symbol:   class ObjectStreamClassReflection
[error]   location: class SerializationDebugger
[error] /home/jenkins/workspace/NewSparkPullRequestBuilder/core/target/java/org/apache/spark/serializer/SerializationDebugger.java:22: error: class SerializationDebugger is already defined in package org.apache.spark.serializer
[error]   static private  class SerializationDebugger {
[error]                   ^
[error] /home/jenkins/workspace/NewSparkPullRequestBuilder/external/kafka-0-10-token-provider/target/java/org/apache/spark/kafka010/KafkaDelegationTokenTest.java:6: error: illegal combination of modifiers: public and private
[error]   private  class KafkaJaasConfiguration extends javax.security.auth.login.Configuration {
[error]            ^
...
```

It's from javadoc'ing the Java code auto-generated from Scala, and this kind of thing is kind of a known problem; I thought we didn't run javadoc on this part. I have no idea so far why it came up in running these tests!

**Comment by cloud-fan** [comment]
ok to test

**Comment by HyukjinKwon** [comment]
If it still fails, let me fix it against this branch since I am kind of used to it.

**Comment by HyukjinKwon** [comment]
Will take a look late today (KST)

**Comment by srowen** [comment]
Note that these errors also appear in 'normal' builds; they're just warnings. I still have no idea why they are rendered as errors due to this change. It doesn't touch the build. I could try separately updating sbt-unidoc, but don't think that's the issue. Still looking...

**Comment by HyukjinKwon** [comment]
Yes .. I think we talked few times long long ago @srowen. I made some analysis on https://github.com/gatorsmile/spark/pull/5

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by cloud-fan** [comment]
retest this please

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**